src/SignalR/clients/java/signalr/test/src/main/java/com/microsoft/signalr/HubConnectionTest.java

```diff
@@ -2783,6 +2783,26 @@ public void negotiateSentOnStart() {
         assertEquals("http://example.com/negotiate?negotiateVersion=1", sentRequests.get(0).getUrl());
     }
 
+    @Test
+    public void closeWithPendingNegotiate() {
+        TestHttpClient client = new TestHttpClient()
+                .on("POST", (req) -> Single.just(new HttpResponse(404, "", TestUtils.emptyByteBuffer)).delay(200, TimeUnit.MILLISECONDS));
```

Comment:
Is it possible to set the Single's value after calling hubConnection.stop so we don't rely on delays as much?

Reply:
It is, that is a good call. Made the change, and also checked that the bug would have been caught with the changed test.